documentation/src/docs/watermarking.md

```diff
@@ -0,0 +1,121 @@
+
+`void commit(long timestamp) throws TxnFailedException;`
+
+For example:
+
+    Transaction<EventType> txn = writer.beginTxn();
+      //... write events to transaction.
+    txn.commit(txnTimestamp);
+
+### Automatically attach wall clock time
+
+The `automaticallyNoteTime` option when a  Writer is created configures the   writer to automatically attach a wall clock time notation to every event. This option essentially attaches the ingest time to each event. With this option turned on, no other calls to note the time are required.
+
+Here is the call to configure automatic timestamps on events:
+
+    EventStreamWriter<EventType> writer = clientFactory.createEventWriter(stream, serializer,
+                EventWriterConfig.builder().automaticallyNoteTime(true).build());
+
+
+## How Pravega Readers request watermark windows  
+
+Multiple Pravega Readers working in parallel each have different current locations in the stream. The Controller coordinates the Reader positions and can return a window that indicates where in the stream all of the readers are.  
+
+The following Reader API method requests the current watermark window.
+
+`TimeWindow window = reader.getTimeWindow();`
+
+The `TimeWindow` is returned as:
+
+    @Data
+    public class TimeWindow {
+        private final long lowerTimeBound;
+        private final long upperTimeBound;
+    }
+
+where:
+* `lowerTimeBound` of the  window is the watermark. The Controller asserts that all readers are done reading all events earlier than the watermark.
+*  `upperTimeBound` is an arbitrary assignment that can help applications keep track of a moving window of time.
+
+The lower bound is most important. The lower bound is labeled the watermark. It takes into account the position of all of the readers and represents the time before which all events are "guaranteed" to have been read. Events can arrive out of order but the watermark would indicate which window each event aggregrates into. At some point, a window needs to be declared closed, so that processing can occur.  Events can also arrive late, after its window has been closed. The action on such an occurrence is up to the application.
+
+If a reader asks for a time window from a stream whose writers are not generating time stamps, then null is returned.  
+
+The TimeWindow reflects the current position in the stream. It can be called following every `readNextEvent()` call if desired, or just periodically to provide support for grouping events into windows.
+
+### Reader Example
+
+Here is an example of a Reader application that...
+
+   >>>>MISSING: I think a reader example that simply requests a window, and does something useful as a result (computes an average or something) and prints a message stating average accurate up to <watermarked lower bound>. I don't understand what the example in the blog is doing. It contains some Flink calls, I think, which complicates the concepts.
```

Comment:
I don't think we should base the example on the flink one. Instead let's describe a specific use case at the top and use it to guide the rest of the document